src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,206 @@
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String insertRequest = "INSERT INTO cars (title, manufacturer_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement insertStatement
+                        = connection.prepareStatement(insertRequest,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            insertStatement.setString(1, car.getTitle());
+            insertStatement.setLong(2, car.getManufacturer().getId());
+            insertStatement.executeUpdate();
+            ResultSet resultSet = insertStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                Long id = resultSet.getObject(1, Long.class);
+                car.setId(id);
+            }
+            return car;
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Couldn't create a car: " + car, throwables);
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT c.id, c.title, c.manufacturer_id "
+                + "FROM cars c JOIN manufacturer m ON c.manufacturer_id = m.id"
+                + " WHERE id = ? AND is_deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement getDriverStatement
+                                = connection.prepareStatement(query)) {
+            getDriverStatement.setLong(1, id);
+            ResultSet resultSet = getDriverStatement.executeQuery();
+            Car car = null;
+            if (resultSet.next()) {
+                car.setDrivers(getDriversById(id, connection));
+                car.setId(id);
+                car.setManufacturer(getManufacturerBySet(resultSet));
+                car.setTitle(resultSet.getString(2));
+            }
+            return Optional.ofNullable(car);
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get a car by id " + id, throwable);
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT c.id, c.title, c.manufacturer_id "
+                + "FROM cars c JOIN manufacturer m ON c.manufacturer_id = m.id"
+                + " WHERE is_deleted = false";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement getDriverStatement
+                                = connection.prepareStatement(query)) {
+            ResultSet resultSet = getDriverStatement.executeQuery();
+            Car car = null;
+            while (resultSet.next()) {
+                Long id = resultSet.getObject(1, Long.class);
+                car.setDrivers(getDriversById(id, connection));
```

Comment:
let's call this method after `catch` block for each of the cars in our `cars` list
the purpose is not to open a connection inside another connection